public/script.js

```diff
@@ -0,0 +1,249 @@
+import { getCategories, getToken, getPlaylists, search } from './api.js';
+
+// html элемент, в котором находится основной контент на странице
+const content = document.querySelector('.content');
+// поисковая строка
+const input = document.querySelector('.header__search');
+
+/** 
+ * Функция для создания заготовки раздела страницы
+ * @returns html элемент section, состоящего из h2 - заголовка и div - контейнера для хранения элементов
+ */
+function createSection() {
+  let section = document.createElement('section');
+  section.className = 'section';
+  const template = `
+  <h2 class="section__header"></h2>
+  <div class="items"></div>
+  `;
+  section.insertAdjacentHTML('beforeend', template);
+  return section;
+}
+
+/**
+ *  Функция для создания элемента раздела страницы на основе переданного объекта 
+ * @returns ссылку, состоящую из изображения и подписи
+*/
+function objectToHtmlElement(obj) {
+  const a = document.createElement('a');
+  a.tabIndex = 0;
+  a.classList = 'item link footer__item-link';
+  a.href = obj.external_urls.spotify;
+  const image_url = obj.images.length == 0 ? "photos/default.jpg" : obj.images[0].url;
+  const template = `
+  <img class="item-image" src="${image_url}" alt="image" width="220" height="220">
+  <p class="item-header">${obj.name}</p>
+  `;
+  a.insertAdjacentHTML('beforeend', template);
+  return a;
+}
+
+/**
+*  Функция для создания раздела с плейлистами 
+* @param {object[]} playlists - массив плейлистов
+* @param {string} categoryName - название категории плейлистов
+* @returns раздел, содержащий название категории и ссылки на плейлисты из категории
+*/
+function playlistsToHtmlElement(playlists, categoryName) {
+  let playlistSection = createSection().cloneNode(true);
```

Comment:
Эта переменная никак не перезаписывает свое значение в дальнейшем. Тут правильнее было бы сделать ее const
```suggestion
  const playlistSection = createSection().cloneNode(true);
```